layer.md

```diff
@@ -1,63 +1,129 @@
-# Creating an Image Filesystem Changeset
+# Filesystem Layer Changeset
 
-An example of creating an Image Filesystem Changeset follows.
+A layer changeset is the distributable difference between filesystems.
+A layer changeset is not directly extracted, but applied to a filesystem.
 
-An image root filesystem is first created as an empty directory.
-Here is the initial empty directory structure for a changeset using the randomly-generated directory name `c3167915dc9d` ([actual layer DiffIDs are generated based on the content](#id_desc)).
+## Distributable Format
+
+Layer Changesets for the [mediatype](./media-types.md) `application/vnd.oci.image.layer.tar+gzip` MUST be packaged in [tar archive][tar-archive].
+Layer Changesets for the [mediatype](./media-types.md) `application/vnd.oci.image.layer.tar+gzip` MUST NOT include duplicate entries for file paths in the resulting [tar archive][tar-archive].
+
+## Change Types
+
+Types of changes that can occur in a changeset are:
+
+* Additions
+* Modifications
+* Removals
+
+Additions and Modifications are reflected the same in the changeset tar archive.
+
+Removals are reflected using "[whiteout](#whiteouts)" file entries (See [Representing Changes](#representing-changes)).
+
+### File Attributes
+
+File attributes that are provided for Additions and Modifications include:
+
+* Modification Time (`mtime`)
+* User ID (`uid`)
+ * User Name (`uname`) is secondary to `uid`
+* Group ID (`gid `)
+ * Group Name (`gname`) is secondary to `gid`
+* Mode (`mode`)
```

Comment:
well, mode is not the same as the file type. Perhaps file type ought to be a field, because if `foo` changes from a regular file into a FIFO, that is a modification.

As for hardlink, perhaps but could be limited by underlying fs support. As for symlink, yes, this reference is effectively the content of the file.